pkg/controller/periodic/periodic.go

```diff
@@ -311,6 +312,8 @@ func (c *Controller) periodicTrigger(stopCh <-chan struct{}) {
 	}
 }
 
+const gatheringLimit = 3
```

Comment:
This value was used for testing, and we should still discuss whether it should be configurable or not. If not, we need to decide what value should be used here.